packages/x-adapter/src/mapper.types.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Transforms the given entry parameter.
+ *
+ * @public
+ */
+export type Mapper<From, To> = (from: Readonly<From>, context: MapperContext) => To;
+
+/**
+ * The context is an object that can be used to share information between the invocation of
+ * different mappers or the process that they are being executed on.
+ *
+ * @public
+ */
+export interface MapperContext {
+  /**
+   * When the mapper is used to transform a response, the request that triggered it.
+   */
+  request?: Record<string, unknown>;
+  /**
+   * When the mapper is used to transform a request or a response, the endpoint that
+   * was or is going be used.
+   */
+  endpoint?: string;
+  /**
+   * If multiple mappers are taking charge of mapping the same property, the partial result of
+   * applying them to it.
+   */
+  to?: unknown;
```

Comment:
I decided to move this to the context to hide it a little bit. I'm not sure whether we are going to need the `to` with the new  schema mappers. Relying on the previous mapper result is a code smell, an indirect dependency that you don't have control of, so I would try to avoid it in the core adapter implementation.

In the case we need it we can always cast it, but (for once) I would rather not complicate types with unnecessary generics just to type scenarios that should not happen.